http/src/http.ml

```diff
@@ -755,6 +755,15 @@ module Request = struct
     | `GET | `HEAD | `CONNECT | `TRACE -> `No
     | `DELETE | `POST | `PUT | `PATCH | `OPTIONS | `Other _ ->
         Transfer.has_body req.encoding
+
+  let[@warning "-3"] make ?(meth = `GET) ?(version = `HTTP_1_1)
```

Comment:
This seems a regression forcing the scheme to be `None` and ignoring all the changes in https://github.com/mirage/ocaml-cohttp/blob/master/cohttp/src/request.ml#L59 

Why does the warning 3 happen? You can get rid of it just by usign Cohttp.Private.Make instead of Cohttp.Make

Reply:
Ah, I wasn't aware there was make in `cohttp`.  That function is what I was after. 

I am using `http` from eio client/server so am not using `cohttp` package. Should the make in https://github.com/mirage/ocaml-cohttp/blob/master/cohttp/src/request.ml#L59 be moved/copied to `http` package itself since it is useful to users who consume `http` only?

Reply:
Ping @rgrinberg

Reply:
@mseri Ah sorry, I hadn't refreshed the browser window when you made the comment above so didn't see it before posing mine. I have now made the `make` even more simple, i.e. not even trying to guess the `encoding` since that attribute is marked as deprecated.

Reply:
Like we do for response: https://github.com/mirage/ocaml-cohttp/blob/b2c3198658fda65cab7f5275bcd2ea329a146b9f/http/src/http.ml#L783

Reply:
Done.